solidity/test/LPRewardsTest.js

```diff
@@ -402,3 +406,120 @@ describe("LPRewards", () => {
     await wrappedToken.approve(lpRewards.address, amount, {from: staker})
   }
 })
+
+describe("LPRewardsTBTCSaddle", () => {
+  const tokenDecimalMultiplier = web3.utils.toBN(10).pow(web3.utils.toBN(18))
+
+  const keepTokenOwner = accounts[1]
+  const lpTokenOwner = accounts[2]
+
+  const lpRewardsOwner = accounts[3]
+
+  const stakerEOA = accounts[4]
+  const thirdParty = accounts[5]
+
+  let keepToken
+  let lpToken
+  let lpRewards
+  let lpRewardsStaker
+
+  before(async () => {
+    keepToken = await KeepToken.new({from: keepTokenOwner})
+    lpToken = await WrappedToken.new({from: lpTokenOwner})
+    lpRewards = await LPRewardsTBTCSaddle.new(
+      keepToken.address,
+      lpToken.address,
+      {from: lpRewardsOwner}
+    )
+    lpRewardsStaker = await LPRewardsStaker.new(
+      lpToken.address,
+      lpRewards.address
+    )
+
+    const keepRewardsAmount = web3.utils.toBN(2016).mul(tokenDecimalMultiplier)
+    await lpRewards.setRewardDistribution(keepTokenOwner, {
+      from: lpRewardsOwner,
+    })
+    await keepToken.approve(lpRewards.address, keepRewardsAmount, {
+      from: keepTokenOwner,
+    })
+    await lpRewards.notifyRewardAmount(keepRewardsAmount, {
+      from: keepTokenOwner,
+    })
+  })
+
+  beforeEach(async () => {
+    await createSnapshot()
+  })
+
+  afterEach(async () => {
+    await restoreSnapshot()
+  })
+
+  describe("setGated", () => {
+    it("can be called by owner", async () => {
+      await lpRewards.setGated(false, {from: lpRewardsOwner})
+      // ok, did not revert
+    })
+
+    it("can not be called by non-owner", async () => {
+      await expectRevert(
+        lpRewards.setGated(false, {from: thirdParty}),
+        "Ownable: caller is not the owner"
+      )
+    })
+  })
+
+  describe("stake", () => {
+    it("allows anyone to stake in a non-gated state", async () => {
```

Comment:
Can we split test cases so we have them separated?
- in the non-gated state allows an externally owned account to stake
- in the non-gated state allows a contract to stake
- in the gated state allows an externally owned account to stake
- in the gated state rejects a contract to stake

Reply:
https://github.com/keep-network/keep-ecdsa/pull/675/commits/9f9011e0c707ca64b74b8348d8ead07b90a9a28e